openedx/core/lib/cache_utils.py

```diff
@@ -151,7 +151,7 @@ def __get__(self, obj, objtype):
 
 def zpickle(data):
     """Given any data structure, returns a zlib compressed pickled serialization."""
-    return zlib.compress(pickle.dumps(data, pickle.HIGHEST_PROTOCOL))
+    return zlib.compress(pickle.dumps(data, 2))  # Keep this constant as we upgrade from python 2 to 3.
```

Comment:
I think we technically also need to [set an encoding](https://github.com/edx/edx-platform/pull/21794/files#diff-dbcfaa375919d30c51526ad44f38180fR242-R245) on the `loads()` side under Python 3:

* https://docs.python.org/3/library/pickle.html#pickle.loads
* https://docs.python.org/3/library/pickle.html#pickle.Unpickler